addons/voice/org.openhab.voice.gtts/src/main/java/org/openhab/voice/gtts/internal/GoogleCloudAPI.java

```diff
@@ -0,0 +1,312 @@
+                googleClient = TextToSpeechClient.create(settings);
+                initialized = true;
+                initVoices();
+            } catch (Exception e) {
+                logger.error("Error initializing the service using {}", keyFile.getAbsoluteFile(), e);
+                initialized = false;
+            }
+
+            //create cache folder
+            cacheFolder = new File(keyFile.getParentFile(), "cache");
+            if (!cacheFolder.exists()) {
+                //noinspection ResultOfMethodCallIgnored
+                cacheFolder.mkdirs();
+            }
+            logger.info("Using cache folder {}", cacheFolder.getAbsolutePath());
+        } else {
+            googleClient = null;
+            voices.clear();
+        }
+    }
+
+    /**
+     * Loads supported audio formats
+     *
+     * @return Set of audio formats
+     */
+    Set<String> getSupportedAudioFormats() {
+        Set<String> formats = new HashSet<>();
+        AudioEncoding[] values = AudioEncoding.values();
+        for (AudioEncoding c : values) {
+            if (c != AudioEncoding.AUDIO_ENCODING_UNSPECIFIED &&
+                    c != AudioEncoding.UNRECOGNIZED) {
+                formats.add(c.toString());
+            }
+        }
+        return formats;
+    }
+
+    /**
+     * Supported locales.
+     *
+     * @return Set of locales
+     */
+    Set<Locale> getSupportedLocales() {
+        return Collections.unmodifiableSet(voices.keySet());
+    }
+
+    /**
+     * Supported voices for locale.
+     *
+     * @param l Locale
+     * @return Set of voices
+     */
+    Set<GoogleTTSVoice> getVoicesForLocale(Locale l) {
+        Set<GoogleTTSVoice> set = voices.get(l);
+        return set != null ? Collections.unmodifiableSet(set) : Collections.EMPTY_SET;
+    }
+
+    /**
+     * Google API call to load locales and voices.
+     */
+    private void initVoices() {
+        if (googleClient != null) {
+            voices.clear();
+            ListVoicesResponse resp = googleClient.listVoices(""); //without language code to get all the voices
+            for (Voice v : resp.getVoicesList()) {
+                for (int i = 0; i < v.getLanguageCodesCount(); i++) {
+                    String languageCode = v.getLanguageCodes(i);
+                    Locale l = Locale.forLanguageTag(languageCode);
+                    Set<GoogleTTSVoice> localVoices;
+                    if (!voices.containsKey(l)) {
+                        localVoices = new HashSet<>();
+                        voices.put(l, localVoices);
+                    } else {
+                        localVoices = voices.get(l);
+                    }
+                    localVoices.add(new GoogleTTSVoice(l, v.getName(), v.getSsmlGenderValue()));
+                }
+            }
+        } else {
+            logger.error("Google client is not initialized!");
+        }
+    }
+
+    /**
+     * Converts OpenHAB audio format to google parameters.
```

Comment:
OpenHAB -> openHAB
google -> Google

Please also make sure the right capitalization is used for these brands in other comments/documentation.